packages/request-state/src/runtime/middleware.ts

```diff
@@ -0,0 +1,22 @@
+import type { ReadableStream } from 'node:stream/web';
+import { defineMiddleware } from 'astro/middleware';
+import { collectState } from './serverState.js';
+
+export const onRequest = defineMiddleware(async (_, next) => {
+	const { getState, result } = await collectState(next);
+
+	async function* render() {
+		for await (const chunk of result.body as ReadableStream<ArrayBuffer>) {
```

Comment:
I think you need to check if the response is html?

Reply:
Good catch